subsys/bluetooth/host/hci_core.c

```diff
@@ -1268,7 +1268,9 @@ static void conn_auto_initiate(struct bt_conn *conn)
 
 	if (IS_ENABLED(CONFIG_BT_AUTO_PHY_UPDATE) &&
 	    !atomic_test_bit(conn->flags, BT_CONN_AUTO_PHY_COMPLETE) &&
-	    BT_FEAT_LE_PHY_2M(bt_dev.le.features)) {
+	    BT_FEAT_LE_PHY_2M(bt_dev.le.features) &&
+	    !(atomic_test_bit(conn->flags, BT_CONN_AUTO_FEATURE_EXCH) &&
+	      !BT_FEAT_LE_PHY_2M(conn->le.features))) {
```

Comment:
I'm having a bit of a hard time wrapping my head around this logic. Would the following make any more sense:
```
        (!atomic_test_bit(conn->flags, BT_CONN_AUTO_FEATURE_EXCH) ||
         BT_FEAT_LE_PHY_2M(conn->le.features))) {
```

I.e. remove the outer negation, negate the internal terms and change the logical AND to a logical OR.

Reply:
Yeah, it looks better. Updated.